src/test_creation/modules/workflow/parse.py

```diff
@@ -7,28 +7,51 @@
 from .response import EvaluationResponse
 from ..mixins import ExportableMixin
 
+from modules.code_analyzer.repo import Repository
+
 
 class ResponseParser(ExportableMixin):
-    def __init__(self, response: EvaluationResponse):
+    def __init__(self, response: EvaluationResponse, respository: Repository = None):
+        # FIXME: respository is required to extract the line numbers for functions
+        #        I added an optional argument "respository" here, can't think of any better way to handle it yet
         self.response = response
         self.evaluation_report = None
+        self.repository = respository
+        self.items = []
+
+    def _parse_items(self):
+        items = []
+        for result in self.response.call_results:
+            response = result.parsed_response['results']
+            for item in response:
+                fp = result.files_evaluated[0] # FIXME: it might fail if the evaluation is on multiple files
+                item['File Path'] = fp
+                if self.repository:
+                    item['lineno'] = [self.repository.ffl_map[fp][func] for func in item['Functions']]
+                else:
+                    item['lineno'] = []
+                item['Line Numbers'] = [
+                    f"[{lineno}]({self.repository._get_git_direct_link(self.repository._get_relative_path(fp), lineno)})"
```

Comment:
I think we should add a check in case that `self.repository._get_git_direct_link` is called when gadd git metadata is empty, it probably will return something like `https://///blob//{file-name}.py` if not checked.

Reply:
resolved in [2bf68a7](https://github.com/UBC-MDS/test-creation/pull/107/commits/2bf68a7a40384f17d5131482561774edb927d1c7)